torch/cuda/__init__.py

```diff
@@ -79,13 +79,24 @@ def _is_compiled() -> bool:
     r"""Returns true if compile with CUDA support."""
     return hasattr(torch._C, '_cuda_getDeviceCount')
 
+def _nvml_based_avail() -> bool:
+    return os.getenv('PYTORCH_NVML_BASED_CUDA_CHK') == '1'
+
 def is_available() -> bool:
     r"""Returns a bool indicating if CUDA is currently available."""
     if not _is_compiled():
         return False
-    # This function never throws and returns 0 if driver is missing or can't
-    # be initialized
-    return torch._C._cuda_getDeviceCount() > 0
+    if _nvml_based_avail():
+        # The user has set an env variable to request this availability check that attempts to avoid fork poisoning by
+        # using NVML at the cost of a weaker CUDA availability assessment. Note that if NVML discovery/initialization
+        # fails, this assessment fails back to the default CUDA Runtime API assessment (`cudaGetDeviceCount`)
```

Comment:
```suggestion
        # fails, this assessment falls back to the default CUDA Runtime API assessment (`cudaGetDeviceCount`)
```
Was it mean to say "fall back"? not sure

Reply:
good suggestion, falls back is a bit better I agree.